### PR TITLE
chore: minor tweaks.

### PR DESCRIPTION
**Pull request by jcchavezs**
Just minor tweaks. Feel free to discard those you don't want.

Ping @mchandramouli

### CODE REVIEW

File: httpd/simpleserver.go [hunk 1]
```diff
@@ -22,41 +22,40 @@ type SimpleServer interface {
...
-	conf Conf
+	conf   Conf
 	server *http.Server
-	mux *http.ServeMux
+	mux    *http.ServeMux
 }
 
-func (simpleServer simpleServerImpl) Port() int {
-	return simpleServer.conf.Port
+func (s *simpleServerImpl) Port() int {
```
Comment (jcchavezs) [inline httpd/simpleserver.go]: usually receivers are short.

File: main.go [hunk 2]
```diff
@@ -2,47 +2,56 @@ package main
...
 	var err error
-	if err = addRoutes(simpleServer, conf); err == nil {
-		if err = startHttpsServer(simpleServer); err == nil {
-			glog.Infof("SimpleServer listening in port %v", simpleServer.Port())
-			wait(func() {
-				glog.Infof("Shutting down initiated")
-				simpleServer.Shutdown()
-			})
-			return
+	defer func() {
```
Comment (jcchavezs) [inline main.go]: This simple sugar allows us to do early termination instead of nested ifs.

File: main.go [hunk 3]
```diff
@@ -2,47 +2,56 @@ package main
...
+		simpleServer.Shutdown()
+	})
+	return
 }
 
 func addRoutes(simpleServer httpd.SimpleServer, conf config) error {
 	mutator, err := routes.NewMutatorController(conf.sideCarConfigFile)
-	if mutator != nil {
-		simpleServer.AddRoute("/mutate", mutator.Mutate)
+	if err != nil {
```
Comment (jcchavezs) [inline main.go]: Usually validity of the output are checked on the errors instead of returned objects.

File: main.go [hunk 4]
```diff
@@ -58,18 +67,11 @@ func readConfig() config {
...
-		returnErr = err
-	case <-time.After(5 * time.Second):
-	}
-
-	close(errs)
-	return returnErr
+	errC := make(chan error, 1)
+	defer close(errC)
+	simpleServer.Start(errC)
+	err := <-errC
```
Comment (jcchavezs) [inline main.go]: This operation will block until you receive an error through the channel. `select` is not necessary as you don't do anything on the time.After.

File: main.go (hunk 2 above)
Comment (mchandramouli) [reply main.go]: That is neat. Thanks

File: main.go (hunk 4 above)
Comment (mchandramouli) [reply main.go]: I didnt want to block forever. After starting the server, I want to return back to main where it waits for os.signal for SIGTERM

File: main.go (hunk 3 above)
Comment (mchandramouli) [reply main.go]: Yes. I had it like that then I changed to return the err returned to make it simple.  I guess, it is possible have an err even if mutator is not nil. so it make sense to check err.

File: httpd/simpleserver.go (hunk 1 above)
Comment (mchandramouli) [reply httpd/simpleserver.go]: Thanks. Is there a reason why you changed them to be pointers? Curious.

File: main.go [hunk 5]
```diff
@@ -57,25 +59,27 @@ func readConfig() config {
 	return conf
 }
 
-func startHttpsServer(simpleServer httpd.SimpleServer) error {
-	errs := make(chan error, 1)
-	simpleServer.Start(errs)
+func startHttpServerAndWait(simpleServer httpd.SimpleServer, callback func()) error {
```
Comment (mchandramouli) [inline main.go]: @jcchavezs see if this implementation make sense. I combined start and wait into a single method. After starting, either I listen to errors or os signal to terminate - on either one, I come out of the blocking select and return.

**Pull request merged**